### PR TITLE
Polish: instance methods should not write to "static" fields

### PR DESCRIPTION
**Pull request by igor-suhorukov**

### CODE REVIEW

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/ConditionEvaluationDeltaLoggingListener.java [hunk 1]
```diff
@@ -33,27 +33,31 @@
...
+	private static final Log logger = LogFactory.getLog(ConditionEvaluationDeltaLoggingListener.class);
 
 	private static ConditionEvaluationReport previousReport;
 
 	@Override
 	public void onApplicationEvent(ApplicationReadyEvent event) {
+		logConditionEvaluationDelta(event);
+	}
+
+	private static synchronized void logConditionEvaluationDelta(ApplicationReadyEvent event) {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/ConditionEvaluationDeltaLoggingListener.java]: Why is this now `synchronized`?

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/HateoasObjenesisCacheDisabler.java [hunk 2]
```diff
@@ -46,26 +46,26 @@ public void afterPropertiesSet() {
 		disableCaching();
 	}
 
-	private void disableCaching() {
+	private static synchronized void disableCaching() {
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/autoconfigure/HateoasObjenesisCacheDisabler.java]: Why is this now `synchronized`?

**Comment by igor-suhorukov** [comment]
@wilkinsona thanks for review. I fix it

**Comment by philwebb** [comment]
I'm not sure I totally understand the logic behind this one. Looking at the `HateoasObjenesisCacheDisabler` change for example, the static field is still being updated by an instance method. It's just now we've made a few more methods static to do it.

**Comment by igor-suhorukov** [comment]
@philwebb thank you for feedback! You are right. Main idea was separate static methods from instance methods. Do you know more correct approach to save state and compare it between context restart?

**Comment by wilkinsona** [comment]
I’m not sure that there’s anything incorrect about the current approach. Being static just affects the scope of the variable and that ClassLoader-wide scope is exactly what we want. There’s no harm in instance methods writing to that scope, particularly in this case where we know there’s only a single instance at a time.

**Comment by philwebb** [comment]
I'm inclined to agree with Andy. I think for these two example we're best off with the code as it is.

**Comment by philwebb** [comment]
Thanks anyway for the suggestion.

**Comment by igor-suhorukov** [comment]
@philwebb @wilkinsona thank you for your time

**Pull request closed without merging**